File: YSDB/MSDB/Models/quality_score.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MSDB.Models
{
    public class quality_score : tablebase
    {
        [Key]
        public int id { get; set; }

        public int companyId { get; set; }

        public int? rsProjectId { get; set; }

        public int year { get; set; }

        public DateTime month { get; set; }

        ///<summary>
        ///打分次数 超过三次（包含第三次），所有得分为0
        ///</summary>
        public int scoreCount { get; set; }

        ///<summary>
        /// 工程师ID
        ///</summary>
        public int engineerId { get; set; }

        ///<summary>
        /// 当次总分
        ///</summary>
        [Column(TypeName = "decimal(5,2)")]
        public decimal totalScore { get; set; }

        ///<summary>
        /// 平均分(当scoreCount大于等于3时，该值为0)
        ///</summary>
        [Column(TypeName = "decimal(5,2)")]
        public decimal avgScore { get; set; }

        ///<summary>
        /// 权重
        ///</summary>
        [Column(TypeName = "decimal(5,3)")]
        public decimal weight { get; set; }
        ///<summary>
        /// 总权重
        ///</summary>
        [Column(TypeName = "decimal(5,3)")]
        public decimal totalWeight { get; set; }

        ///<summary>
        /// 通过情况
        ///</summary>
        [Column(TypeName = "decimal(5,2)")]
        public decimal passRate { get; set; }

        ///<summary>
        /// 类型：0 立项报告
        ///</summary>
        public int type { get; set; }

        public bool isFinal { get; set; }

        public int? monthsId { get; set; }

        ///<summary>
        ///提交时间
        ///</summary>
        public DateTime submitTime { get; set; }

        ///<summary>
        ///审核时间
        ///</summary>
        public DateTime auditTime { get; set; }
    }
}

File: MS/src/main/java/com/yskc/ms/entity/ms/QualityScore.java
package com.yskc.ms.entity.ms;

import com.baomidou.mybatisplus.annotations.TableId;
import com.baomidou.mybatisplus.annotations.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.yskc.ms.entity.ms.models.MsBaseEntity;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.Date;

/**
 * @DateTime: 2022/5/13 9:32
 * @Description:审核评分entity
 * @author: hsx
 */
@TableName("quality_score")
public class QualityScore extends MsBaseEntity implements Serializable  {

    private static final long serialVersionUID = 6551502756636529888L;

    @TableId
    private Integer id;

    private Integer companyId;

    private Integer rsProjectId;

    private Integer year;

    private Date month;

    private Integer scoreCount;

    private Integer engineerId;

    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private BigDecimal totalScore;

    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private BigDecimal avgScore;

    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private BigDecimal weight;

    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private BigDecimal passRate;

    private Integer type;

    private Integer monthsId;

    private Boolean isFinal;

    private Date submitTime;

    private Date auditTime;

    private BigDecimal totalWeight;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getCompanyId() {
        return companyId;
    }

    public void setCompanyId(Integer companyId) {
        this.companyId = companyId;
    }

    public Integer getRsProjectId() {
        return rsProjectId;
    }

    public void setRsProjectId(Integer rsProjectId) {
        this.rsProjectId = rsProjectId;
    }

    public Integer getYear() {
        return year;
    }

    public void setYear(Integer year) {
        this.year = year;
    }

    public Date getMonth() {
        return month;
    }

    public void setMonth(Date month) {
        this.month = month;
    }

    public Integer getScoreCount() {
        return scoreCount;
    }

    public void setScoreCount(Integer scoreCount) {
        this.scoreCount = scoreCount;
    }

    public Integer getEngineerId() {
        return engineerId;
    }

    public void setEngineerId(Integer engineerId) {
        this.engineerId = engineerId;
    }

    public BigDecimal getTotalScore() {
        return totalScore;
    }

    public void setTotalScore(BigDecimal totalScore) {
        this.totalScore = totalScore;
    }

    public BigDecimal getAvgScore() {
        return avgScore;
    }

    public void setAvgScore(BigDecimal avgScore) {
        this.avgScore = avgScore;
    }

    public BigDecimal getWeight() {
        return weight;
    }

    public void setWeight(BigDecimal weight) {
        this.weight = weight;
    }

    public BigDecimal getPassRate() {
        return passRate;
    }

    public void setPassRate(BigDecimal passRate) {
        this.passRate = passRate;
    }

    public Integer getType() {
        return type;
    }

    public void setType(Integer type) {
        this.type = type;
    }

    public Integer getMonthsId() {
        return monthsId;
    }

    public void setMonthsId(Integer monthsId) {
        this.monthsId = monthsId;
    }

    public Boolean getFinal() {
        return isFinal;
    }

    public void setFinal(Boolean aFinal) {
        isFinal = aFinal;
    }

    public Date getSubmitTime() {
        return submitTime;
    }

    public void setSubmitTime(Date submitTime) {
        this.submitTime = submitTime;
    }

    public Date getAuditTime() {
        return auditTime;
    }

    public void setAuditTime(Date auditTime) {
        this.auditTime = auditTime;
    }

    public BigDecimal getTotalWeight() {
        return totalWeight;
    }

    public void setTotalWeight(BigDecimal totalWeight) {
        this.totalWeight = totalWeight;
    }
}
